lib/cookie/cookieJar.ts

```diff
@@ -188,9 +188,28 @@ export interface CreateCookieJarOptions {
 const SAME_SITE_CONTEXT_VAL_ERR =
   'Invalid sameSiteContext option for getCookies(); expected one of "strict", "lax", or "none"'
 
-function getCookieContext(url: unknown): URL {
-  if (url instanceof URL) {
-    return url
+type UrlContext = {
+  hostname: string
+  pathname: string
+  protocol: string
+}
+
+function getCookieContext(url: unknown): UrlContext {
+  if (
+    url &&
+    typeof url === 'object' &&
+    'hostname' in url &&
+    typeof url.hostname === 'string' &&
+    'pathname' in url &&
+    typeof url.pathname === 'string' &&
+    'protocol' in url &&
+    typeof url.protocol === 'string'
```

Comment:
Could be nice to extract this to a `isUrlLike` helper. I also feel like we should skip the `in` checks and just do `typeof` checks. And tell TypeScript to shush, because we're confident in what we're doing here.

Reply:
I don't mind the extra checks seeing as how TypeScript does require them since the object is `unknown`. I'm more confident that the TypeScript compiler knows what it's doing than me most days 😅.

Also, the `getCookieContext` is essentially already a `isUrlLike` helper. The naming is just odd since we're keeping a lot of the legacy function names untouched. Since we don't duplicate this code anwhere else, I don't think it needs to be extracted.